docs/source/daml/intro/3_Data.rst

```diff
@@ -0,0 +1,202 @@
+- ``Date``
+  Stores a date.
+- ``Time``
+  Stores absolute UTC time.
+- ``RelTime``
+  Stores a difference in time.
+
+The below scenario instantiates each one of these types manipulates it where appropriate and tests the result.
+
+.. literalinclude:: daml/3_Data/Native.daml
+  :language: daml
+  :start-after: -- NATIVE_TEST_BEGIN
+  :end-before: -- NATIVE_TEST_END
+
+Despite its simplicity, there are quite a few things to note in this scenario:
+
+1. the ``import`` statements at the top import two packages from the DAML Standard Library, which contain all the date and time related functions we use here. More on packages, imports and the standard library later.
+2. Most of the variables are declared inside a ``let`` block. That's because the ``scenario do`` block expects scenario actions like ``submit`` or ``getParty``. An integer like ``123`` is not an action, it's a pure expression we can evaluate without any ledger. The ``let`` can be thought of as turning variable declaration into an action.
+3. None of the variables have type annotations. That's because DAML is very good at *inferring* types. The compiler knows that ``123`` is an ``Int`` so if you declare ``my_int = 123``, it can infer that ``my_int`` is also an ``Int`` so you don't have to write ``my_int : Int = 123``. You can always add type annotations to aid readability, and sometimes, in case of ambiguity, have to.
+4. The ``assert`` function is an action that takes a boolean value and succeeds with ``True`` and fails with ``False``. Try putting ``assert False`` somewhere in a scenario and see what happens to the scenario result.
+
+With tempates and these native types, it's already possible to write a schema akin to table in relational database. Below, ``Token`` is extended into a simple ``CashBalance``, administered by a party in the role of an accountant.
+
+.. literalinclude:: daml/3_Data/Native.daml
+  :language: daml
+  :start-after: -- CASH_BALANCE_BEGIN
+  :end-before: -- CASH_BALANCE_END
+
+Assembling types
+----------------
+
+There's quite a lot of information on the ``CashBalance`` above and it would be nice to be able to give that data more structure. Fortunately, DAML's type system has a number of ways to assemble these native types into much more expressive structures.
+
+Tuples
+~~~~~~
+
+A common task is to group values in a generic way. Take, for example a key-value pair with a ``Text`` key and an ``Int`` value. In DAML, you could use a two-tuple of type ``(Text, Int)`` to do so. If you wanted to express a coordinate in three dimensions, you could group three ``Decimal`` values using a three-tuple ``(Decimal, Decimal, Decimal)``.
+
+
+.. literalinclude:: daml/3_Data/Tuple.daml
+  :language: daml
+  :start-after: -- TUPLE_TEST_BEGIN
+  :end-before: -- TUPLE_TEST_END
+
+Note how the data can be accessed either through functions ``fst``, ``snd``, ``fst3``, ``snd3``, ``thd3``, or via a dot-syntax with field names ``_1``, ``_2``, ``_3``, etc.
```

Comment:
I think we should mention here what the maximum supported size is.

Reply:
```suggestion
Note how the data can be accessed either through functions ``fst``, ``snd``, ``fst3``, ``snd3``, ``thd3``, or via a dot-syntax with field names ``_1``, ``_2``, ``_3``, etc. DAML supports tuples with up to 20 elements, but accessor functions like ``fst`` are only included for 2- and 3-tuples.
```